examples_utils/notebook_logging/gc_logger.py

```diff
@@ -159,60 +159,68 @@ def __update_payload(cls, output: str, name: str) -> str:
     def __get_notebook_metadata(cls):
         """Get notebook metadata."""
 
-        while True:
-            if cls.LOG_STATE == "DISABLED":
-                return
+        try:
+            while True:
+                if cls.LOG_STATE == "DISABLED":
+                    return
 
-            try:
-                notebook_path = str(ipynbname.path())
-            except:
-                notebook_path = ""
+                try:
+                    notebook_path = str(ipynbname.path())
+                except:
+                    notebook_path = ""
+
+                # Encode and hash
+                notebook_id = os.getenv("PAPERSPACE_NOTEBOOK_ID")
+                salted_id = notebook_id + datetime.now().strftime("%Y-%m-%d")
+                anonymised_notebook_id = base64.urlsafe_b64encode(
+                    hashlib.md5(salted_id.encode("utf-8")).digest()
+                ).decode("ascii")[:16]
 
-            # Encode and hash
-            notebook_id = os.getenv("PAPERSPACE_NOTEBOOK_ID")
-            salted_id = notebook_id + datetime.now().strftime("%Y-%m-%d")
-            anonymised_notebook_id = base64.urlsafe_b64encode(hashlib.md5(salted_id.encode("utf-8")).digest()).decode(
-                "ascii"
-            )[:16]
+                notebook_metadata = {
+                    "notebook_path": notebook_path,
+                    "notebook_repo_id": os.getenv("PAPERSPACE_NOTEBOOK_REPO_ID"),
+                    "notebook_id": anonymised_notebook_id,
+                    "cluster_id": os.getenv("PAPERSPACE_CLUSTER_ID"),
+                }
 
-            notebook_metadata = {
-                "notebook_path": notebook_path,
-                "notebook_repo_id": os.getenv("PAPERSPACE_NOTEBOOK_REPO_ID"),
-                "notebook_id": anonymised_notebook_id,
-                "cluster_id": os.getenv("PAPERSPACE_CLUSTER_ID"),
-            }
+                for key, val in notebook_metadata.items():
+                    cls.__update_payload(val, key)
 
-            for key, val in notebook_metadata.items():
-                cls.__update_payload(val, key)
+                time.sleep(cls._POLLING_SECONDS)
 
-            time.sleep(cls._POLLING_SECONDS)
+        except:
+            pass
```

Comment:
Is there the opportunity to log something through the logger here? Maybe we can have that as a follow up task as it'll need quite a bit of testing.